### DISCUSSION TITLE
status?

### DISCUSSION THREAD

**Discussion by taijiren** [uncomplicate/bayadera#11] [General]
Hi Dragan,

I had some fun today trying to play with https://github.com/blueberry/doing-bayesian-data-analysis-gpu-opencl (ch08).

I just wanted to see if I could make things run without crashing in spite of it being not documented, not released, more polished for opencl and I'm running nvidia rtx 2060, etc. I know none of this is supported, but I thought, let's see how far I can go...

Cloned bayadera and updated deps to clojurecuda 0.15 and neanderthal 0.41 and commons 0.12
Got the libcudadevrt.a library issue figured out (archlinux installs in /opt/cuda...)
Had to modify nvidia-gtx.clj as the compiler doesn't want any gpu-architecture less than "compute_35"
Spent a lot more time getting quil-processing4 to at least open and not crash...
And finally tried to generate a pretty picture from chapter 8, but trying to sample the generated beta distribution fails with "Value out of range for int" errors.

Not asking you to untangle my mess (I put this here and not as an issue). But, I'm curious what your thoughts are on having bayadera workable for cuda in the near term
Is it a lot to keep it in sync with neanderthal? Is it too tied to a specific video card? You did name it nvidia-gtx and not nvidia or cuda.

**Comment by blueberry** [comment]
Bayadera should work with CUDA. It's just that recent CUDA toolkit no longer supports compute_30, but that was minimal requirement, so you can just change it to whatever recent cuda supports (or remove that option completely, since now CUDA has to support the features that were ensured by that option).

The code is pretty general so it should work well with any recent Nvidia GPU...

Quil 4 is out of my control -> use Java 8 and Quil based on Processing 3.

I'll update Bayadera and continue to develop it, but at the time more people in Clojure community are interested in Neanderthal and Deep Learning, so for some more time the priority will be to improve these libraries.

**Reply by taijiren** [reply]
hmm. I didn't think of trying to remove the gpu-architecture option from those commands.
Thank you. I was under the impression that the cuda side was less ready, but if it should work, I will try again tomorrow.

**Reply by taijiren** [reply]
Success!
The primary issue was something about the change you made in commons with `generate-seed`. Throughout bayadera the seed was either assumed to be `int` or being cast with `(int seed)`, but the long value from commons.generate-seed was overflowing. So, since it's an initial seed value and can (I think) safely lose information, I swapped all the calls in bayadera to `(generate-seed)` with `(.intValue (generate-seed))`. This worked, and I finally got a pretty picture.
Not sure if you want a pull request or to address it in some other way (like in the commons library).

**Comment by blueberry** [comment]
Please just post files and line numbers here, and I'll fix this when I update Bayadera to new dependencies.

**Reply by taijiren** [reply]
Ok.

For the cuda update issue...
src/clojure/uncomplicate/bayadera/internal/device/nvidia_gtx.clj <Lines: 619, 623, 627, 631>
can lose the `"-arch=compute_??"` options as you pointed out (unless the `compute_35` one is still important).

For the generate-seed issue, I replaced `(generate-seed)` with `(.intValue (generate-seed))` in...
src/clojure/uncomplicate/bayadera/mcmc.clj <Lines: 78>
src/clojure/uncomplicate/bayadera/internal/impl.clj <Lines: 94, 96>
src/clojure/uncomplicate/bayadera/internal/distribution_library.clj <Lines: 34, 66, 101, 139, 177, 215, 247>
